desc/configuration.py

```diff
@@ -348,6 +340,16 @@ def __init__(
         if "L_lmn" in kwargs:
             self.L_lmn = kwargs.pop("L_lmn")
 
+        # ensure number of field periods agree
+        eq_NFP = self.NFP
+        surf_NFP = self.surface.NFP if hasattr(self.surface, "NFP") else self.NFP
+        axis_NFP = self.axis.NFP
```

Comment:
with the new changes i think this should use self._axis.NFP, since this is checked before setting the initial guess